src/Tools/Extensions.ApiDescription.Server/src/buildMultiTargeting/Microsoft.Extensions.ApiDescription.Server.targets

```diff
@@ -33,8 +32,9 @@
         RemoveProperties="RuntimeIdentifier" />
   </Target>
 
+  <!-- DispatchToInnerBuilds is last target outer Build depends on. -->
   <Target Name="_GenerateOpenApiDocuments"
-      AfterTargets="Build"
+      AfterTargets="DispatchToInnerBuilds"
```

Comment:
Could you instead use `BeforeTargets="Build"`? That keeps from depending on the confusing Dispatch target.

Reply:
My aim here is to run `GenerateOpenApiDocuments` in just one inner build (with a TFM the tool supports) regardless of the number of entries in `$(TargetFrameworks)`.

The inner `GenerateOpenApiDocuments` target has to run _after_ the project is built. (The tool invoked is an "outside man" that executes the application with a different-than-normal entry point.) Is there another option that would be less confusing?

I guess one option would be to change https://github.com/aspnet/AspNetCore/pull/13364/files#diff-b3e35eb5bf39d86aaddd2a229f8cef8cR30 to `Targets="Build;GenerateOpenApiDocuments"` and use `BeforeTargets="Build"` here. Would that run the specific inner build again when `DispatchToInnerBuilds` executes? Would it even work?

Reply:
**tl;dr**: I think I'm trying to say "`BeforeTargets="Build"` is a drop-in substitute for almost all uses of `AfterTargets="Build"`, because `Build` does nothing itself so ordering around it shouldn't matter".

----

Naming targets is really hard. The convention is to make them logical from the dev's perpective, which makes them _super confusing_ from the build maintainer's perspective.

Specifically, targets with simple names run _after_ all of the steps that make them up. So for example, `Build` should be the last target that runs when you do `/t:Build` (or default target). That means that saying `BeforeTargets="Build"` is a good way to say "run this target last thing before saying 'Build' is done". Because of the [details of target ordering](https://docs.microsoft.com/en-us/visualstudio/msbuild/target-build-order?view=vs-2019#determine-the-target-build-order), that will (in the absence of other constraints) run _after_ all the things that `Build` `DependsOn`, giving you the behavior I think you want in this case.

Reply:
OK, I'll update this too